Add pyproject.toml, loosen dependencies

d8ahazard opened the pull request:
Package was making it very difficult to install other packages that depend on things like numpy, laion-clap, pandas, etc.

zqevans commented:
Happy to make some changes to the dependencies to speed things up. My Python build knowledge is a bit lacking, what is the point of having the dependencies listed in both setup.py and pyproject.toml?

I see a few imports are being added here (`accelerate`, `numpy`, `httpcore`), what are those for?

Our flex attention code also needs PyTorch 2.5+, so that version should be bumped up.

d8ahazard commented:
> Happy to make some changes to the dependencies to speed things up. My Python build knowledge is a bit lacking, what is the point of having the dependencies listed in both setup.py and pyproject.toml?
> 
> I see a few imports are being added here (`accelerate`, `numpy`, `httpcore`), what are those for?
> 
> Our flex attention code also needs PyTorch 2.5+, so that version should be bumped up.

I'd be lying if I said there was a strict reason to have the deps specifically spelled out in both files. I'm not an expert on pyproject.toml files or the newer build system, so there may be a more elegant or preferred way to do this. I just added the new file and updated both with the same dependencies so of any existing ci/cd pipelines were using setup.py, it wouldn't break anything.

The extra imports were unintentional - I have a utility set up where I'm dealing with multiple repos like this where I was getting import issues (numpy...lol...I hate numpy), and one of my early scripts got a bit too excited and inserted the *same* deps in all the projects setup files. Everything builds fine, but yeah, totally unneeded.

I've removed these, and bumped the torch version as requested.

zqevans commented:
needing to maintain identical dependencies in two different files sounds like a bug waiting to happen. I'd rather keep pyptoject.toml minimal as it was, I believe it's only in there so I can run the build command to make the pypi package.

torkleyy commented on pyproject.toml:
```diff
@@ -1,3 +1,55 @@
+]
+description = "Training and inference tools for generative audio models from Stability AI"
+readme = "README.md"
+requires-python = ">=3.8"
```
When using uv sync, I needed python 3.10 specified here.

```suggestion
requires-python = ">=3.10"
```
